Sources/_SwiftSyntaxTestSupport/FixItApplier.swift

```diff
@@ -0,0 +1,149 @@
+      .flatMap(\.changes)
+
+    var edits: [Edit] = []
+
+    for change in changes {
+      switch change {
+      case .replace(let oldNode, let newNode):
+        edits.append(
+          Edit(
+            startUtf8Offset: oldNode.position.utf8Offset,
+            endUtf8Offset: oldNode.endPosition.utf8Offset,
+            replacement: newNode.description
+          )
+        )
+
+      case .replaceLeadingTrivia(let token, let newTrivia):
+        edits.append(
+          Edit(
+            startUtf8Offset: token.position.utf8Offset,
+            endUtf8Offset: token.positionAfterSkippingLeadingTrivia.utf8Offset,
+            replacement: newTrivia.description
+          )
+        )
+
+      case .replaceTrailingTrivia(let token, let newTrivia):
+        edits.append(
+          Edit(
+            startUtf8Offset: token.endPositionBeforeTrailingTrivia.utf8Offset,
+            endUtf8Offset: token.endPosition.utf8Offset,
+            replacement: newTrivia.description
+          )
+        )
+      }
+    }
+
+    var source = tree.description
+
+    // As we need to start apply the edits at the end of a source, start by reversing edit
+    // and then sort edits by decrementing start offset. If they are equal then descrementing end offset.
+    //      edits = edits.reversed().sorted(by: { edit1, edit2 in
+    //      if edit1.startUtf8Offset == edit2.startUtf8Offset {
+    //        return edit1.endUtf8Offset > edit2.endUtf8Offset
+    //      } else {
+    //        return edit1.startUtf8Offset > edit2.startUtf8Offset
+    //      }
+    //    })
+
+    while let edit = edits.first {
+      edits = Array(edits.dropFirst())
+
+      let startIndex = source.utf8.index(source.utf8.startIndex, offsetBy: edit.startUtf8Offset)
+      let endIndex = source.utf8.index(source.utf8.startIndex, offsetBy: edit.endUtf8Offset)
+
+      source.replaceSubrange(startIndex..<endIndex, with: edit.replacement)
+
+      edits = edits.compactMap { remainingEdit -> FixItApplier.Edit? in
+        var remainingEdit = remainingEdit
+
+        if remainingEdit.replacementRange.overlaps(edit.replacementRange) {
+          // The edit overlaps with the previous edit. We can't apply both
+          // without conflicts. Apply the one that's listed first and drop the
+          // later edit.
+          return nil
+        }
+
+        // Keep the origianl length as we start to edit the offset.
+        let originalLength = remainingEdit.originalLength
+
+        // If the current is after the edit we want to update, don't change the offset.
+        // If the remaining edit have same start and end position, don't change the start offset.
+        if edit.endUtf8Offset <= remainingEdit.startUtf8Offset {
+          // Take the largest value as we in some cases can end up with a negative offset.
+          remainingEdit.startUtf8Offset = Swift.max(remainingEdit.startUtf8Offset - edit.originalLength + edit.replacementLength, 0)
```

Comment:
Why do we need `Swift.max` here? I think `remainingEdit.startUtf8Offset - edit.originalLength + edit` should always be ≥ 0.

And since I haven’t done a proof in a while and I like them, here’s the proof😉 

```
    remainingEdit.startUtf8Offset - edit.originalLength + edit.replacementLength ≥ 0
<=> remainingEdit.startUtf8Offset - (edit.endOffset - edit.startOffset) + edit.replacementLength ≥ 0
<=> remainingEdit.startUtf8Offset - edit.endOffset + edit.startOffset + edit.replacementLength ≥ 0
<=  remainingEdit.startUtf8Offset - edit.endOffset ≥ 0
<=> remainingEdit.startUtf8Offset ≥ edit.endOffset
(true by the check if the if-condition)
```

I would be fine with adding a precondition for ≥ 0 if you want to.